sycl/plugins/unified_runtime/ur/adapters/cuda/enqueue.cpp

```diff
@@ -429,8 +459,24 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
 
     if (phEvent) {
       Result = RetImplEvent->record();
+      for (auto &MemArg : hKernel->Args.MemObjArgs) {
+        // Telling the ur_mem_handle_t that it will need to wait on this kernel
+        // if it has been written to
+        if (MemArg.Mem->isBuffer() &&
+            MemArg.AccessFlags &
+                (UR_MEM_FLAG_READ_WRITE | UR_MEM_FLAG_WRITE_ONLY)) {
+          ur_cast<ur_buffer_ *>(MemArg.Mem)
+              ->setLastEventWritingToMemObj(RetImplEvent.get());
+        }
+        MemArg.Mem->MemoryMigrationMutex.unlock();
```

Comment:
The same comment as above, when locking.

Reply:
Thanks change made